### ISSUE TITLE
FR:Possible to make Dynamic Links parameters dictionary public

### ISSUE THREAD

**Issue by alihanaktay** [firebase/firebase-ios-sdk#6730]
### Feature proposal

* Firebase Component: Firebase Dynamic Links

In our use case we need to send some parameters(e.g. utm_ campaign, utm_medium etc.) to the backend for checking is any discount available for this campaign. In current developer's can't reach the dynamic links parameters even if received by app. Is it possible to make Dynamic links parameters public?

**Comment by rromanchuk** [comment]
Also  +1 for anything that can be set via console/api  link creation to have public accessors via an instance of `DynamicLink`,  just because i enjoy being lazy and would enjoy not parsing urls for already structured data. 

```swift
let handled = DynamicLinks.dynamicLinks().handleUniversalLink(link) { fdl, error in
  guard let fdl = fdl, let url = fdl.url else { return }
   
  // 😍  😍 😍 😍
  let campaign: String? = fdl.utmCampaign
  let medium: String? = fdl.utmMedium
}
```

**Comment by paulb777** [comment]
Fixed by #7505
